astropy/cosmology/_src/traits/darkenergy.py

```diff
@@ -118,17 +117,15 @@ def de_density_scale(self, z: Quantity | ArrayLike) -> FArray | float:
         # so it is probably a good idea for subclasses to overload this
         # method if an analytic form is available.
         z = aszarr(z)
-        if not isinstance(z, (Number, np.generic)):  # array/Quantity
-            ival = np.array(
-                [quad(self._w_integrand, 0, log(1 + redshift))[0] for redshift in z]
-            )
-            return np.exp(3 * ival)
-        else:  # scalar
-            ival = quad(self._w_integrand, 0, log(z + 1.0))[0]
-            return exp(3 * ival)
+        ival = np.array(
```

Comment:
Again, a direct cast to array is risky. For the scalar case, you can just add `[...]` to make it an array (if you really want that; to me, this feels misplaced).

For the array case, where you have a list, I guess there is less choice. In preparation for the array API, maybe use `np.asarray`, though.